src/main/java/com/hazelcast/aws/impl/DescribeInstances.java

```
@@ -52,15 +52,11 @@ public DescribeInstances(AwsConfig awsConfig, String endpoint) throws IOExceptio
         if (awsConfig == null) {
             throw new IllegalArgumentException("AwsConfig is required!");
         }
-        if (awsConfig.getAccessKey() == null && awsConfig.getIamRole() == null) {

```

we can fail fast here in case they both defined.